src/hotspot/share/opto/replacednodes.cpp

```diff
@@ -208,7 +208,7 @@ void ReplacedNodes::apply(Compile* C, Node* ctl) {
       hash_table_size++;
     }
     // Map from current node to cloned/replaced node
-    ResizeableResourceHashtable<Node*, Node*, AnyObj::RESOURCE_AREA, mtCompiler> clones(hash_table_size, hash_table_size);
+    OrigToNewHashtable clones(hash_table_size, hash_table_size);
```

Comment:
Nice.
Not your problem here. But should there not be a ResouceMark before this hashtable? There is one at the beginning of the function, but we create many of these hashtables in a loop, without any ResourceMarks in between reclaiming the memory...
A but then the hashmaps and stack/to_fix etc would allocate from the ResourceArea, but start at different ResourceMarks... bad idea. Hmm.

Reply:
As discussed offline, we should probably go over all uses of resource allocated things like `Node_Lists`, `Node_Stack` etc. at some point and check if there are missing resource marks.